geoalchemy2/types/__init__.py

```diff
@@ -179,7 +179,8 @@ def process(bindvalue):
     @staticmethod
     def check_ctor_args(geometry_type, srid, dimension, use_typmod, nullable):
         try:
-            srid = int(srid)
+            # passing default SRID is it is NULL from DB
+            srid = int(srid or -1)
         except ValueError:
```

Comment:
```suggestion
        except (ValueError, TypeError):
```
To be more generic

Reply:
I have included this exception in the current commit. Thanks for the suggestion.